network/check.py

```diff
@@ -262,8 +262,17 @@ def _check_linux(self, instance):
                 for ip_version in ['4', '6']:
                     # Call `ss` for each IP version because there's no built-in way of distinguishing
                     # between the IP versions in the output
-                    output, _, _ = get_subprocess_output(["ss", "-n", "-u", "-t", "-a", "-{0}".format(ip_version)], self.log)
-                    lines = output.splitlines()
+                    output_tcp, _, _ = get_subprocess_output(["ss", "-n", "-t", "-a", "-{0}".format(ip_version)], self.log)
+                    output_udp, _, _ = get_subprocess_output(["ss", "-n", "-u", "-a", "-{0}".format(ip_version)], self.log)
```

Comment:
I think this could be simplified by using a second `for` loop iterating on `['tcp', 'udp']` (or having one loop iterating over all 4 combinations of protocols/ip versions) and calling `get_subprocess_output` and `_parse_linux_cx_state` inside the loop. This should allow removing your specific code below that adds a header to each line and modifies the output of the command.

Feel free to change the signature and internal behavior of `_parse_linux_cx_state`, or remove it altogether if it doesn't make sense to use it anymore, it's only used for these `ss` and `netstat` calls.